syntax = "proto3";

package booking;

import "user.proto";
import "schedule.proto";

service BookingService {
    rpc CreateBooking(CreateBookingRequest) returns (BookingResponse);
    rpc CancelBooking(CancelBookingRequest) returns (BookingResponse);
    rpc GetUserBookings(GetUserBookingsRequest) returns (BookingsResponse);
    rpc GetBooking(GetBookingRequest) returns (BookingResponse);
    rpc UserValidated(UserValidationRequest) returns (UserValidationResponse);
}

message CreateBookingRequest {
    string user_id = 1;
    string workout_id = 2;
}

message CancelBookingRequest {
    string booking_id = 1;
}

message GetUserBookingsRequest {
    string user_id = 1;
}

message GetBookingRequest {
    string booking_id = 1;
}

message UserValidationRequest {
    string user_id = 1;
    string workout_id = 2;
}

message UserValidationResponse {
    bool valid = 1;
    string message = 2;
}

message BookingResponse {
    string booking_id = 1;
    string user_id = 2;
    string workout_id = 3;
    string status = 4;
    string created_at = 5;
    string updated_at = 6;
}

message BookingsResponse {
    repeated BookingResponse bookings = 1;
}